src/App.jsx

```diff
@@ -14,45 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getSortedGoods(listOfGoods, { sortType, isReverse }) {
+  let copyOfGoods = [...listOfGoods];
+
+  copyOfGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SORT_FIELD_ALPH:
+        return good1.localeCompare(good2);
+      case SORT_FIELD_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReverse === true) {
+    copyOfGoods = [...copyOfGoods].reverse();
+  }
+
+  return copyOfGoods;
+}
+
+const SORT_FIELD_ALPH = 'alph';
+const SORT_FIELD_LENGTH = 'length';
+
+export const App = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sortType, setSortType] = useState('');
+  const sortedGoods
+  = getSortedGoods(goodsFromServer, { sortType, isReverse });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortType !== SORT_FIELD_ALPH },
+          )}
+          onClick={() => {
+            setSortType(SORT_FIELD_ALPH);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortType !== SORT_FIELD_LENGTH },
+          )}
+          onClick={() => {
+            setSortType(SORT_FIELD_LENGTH);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': isReverse === false },
+          )}
+          onClick={(isReverse === false)
+            ? () => {
+              setIsReverse(true);
+            }
+            : () => {
+              setIsReverse(false);
+            }}
+        >
```

Comment:
```suggestion
          onClick={() => setIsReverse(!isReverse)}
        >
```

can be more simplify